pkg/plugin/v2/api.go

```diff
@@ -196,10 +195,12 @@ func (p *createAPIPlugin) GetScaffolder(c *config.Config) (scaffold.Scaffolder,
 		return nil, fmt.Errorf("unknown pattern %q", p.pattern)
 	}
 
-	return scaffold.NewAPIScaffolder(c, string(bp), res, p.doResource, p.doController, plugins), nil
+	// Create the actual resource from the resource options
+	res := p.resource.NewResource(p.config, p.doResource)
```

Comment:
could it not be inside of NewAPIScaffolder as well?

Reply:
There's a distinction controlled by plugin version between resources created by `NewV1Resource` and `NewResource`, which is why `*Resource` is passed instead of `*Options`. Ditto for webhook plugins.

Reply:
To be fair, scaffolds know project version internally and make decisions based on that information so we could move this into the scaffolder. However I think we should wait until we remove project version 1 and re-assess where we create resources then.